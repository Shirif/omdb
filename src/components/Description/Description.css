.description-container {
  display: flex;
  align-items: center;
  justify-content: center;
  flex-wrap: wrap;
  background-color: #000000;
  padding: 5px;
  color: #d3d3d3;
  font-weight: 300;
  margin: 90px 0;
  box-shadow: 0 -50px 20px #000000d5, 0 50px 20px #000000d5;
}
.description {
  font-size: 0.8em;
  padding: 25px 0;
}
.description h1 {
  margin: 0;
  font-size: 3em;
  font-weight: 300;
  text-align: center;
}
.description p {
  padding-left: 5px;
  margin: 0.4em;

}
.description b {
  color: #757575;
}

.poster-img img {
  width: 320px;
  mask-image: linear-gradient(to right, transparent 1%, black 30%);
}
