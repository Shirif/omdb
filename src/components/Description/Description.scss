.container {
  display: flex;
  align-items: center;
  justify-content: center;
  flex-wrap: wrap;
  background-color: #000000;
  padding: 5px;
  color: #d3d3d3;
  font-weight: 300;
  margin: 20px 0;
  box-shadow: 0 -20px 20px #3d3d3dd5, 0 20px 20px #3d3d3dd5;
}

.description {
  position: relative;
  font-size: 0.8em;
  padding: 25px 0;
  width: 600px;
}

.description h1 {
  margin: 0;
  font-size: 3em;
  font-weight: 300;
  text-align: center;
}

.description p {
  padding-left: 5px;
  margin: 0.4em;
}

.description b {
  color: #757575;
}

.poster img {
  width: 320px;
  height: 470px;
  mask-image: linear-gradient(to right, transparent 1%, black 30%);
}

.close {
  cursor: pointer;
  position: absolute;
  color: #ffffffa2;
  left: 4px;
  top: 4px;
}

.close:hover {
  color: #3fff05;
}
