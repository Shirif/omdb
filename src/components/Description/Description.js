import React from "react";
import { useDispatch, useSelector } from "react-redux";
import { noChangeMovie } from "../../store/actionCreators/action";
import noImg from "../../Images/no_image_available.png";
import "./Description.scss";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { faWindowClose as close } from "@fortawesome/free-regular-svg-icons";

const Description = ({ changeMovie }) => {
  const dispatch = useDispatch({});
  const movies = useSelector((state) => state.movies.fetchMovies);
  const favorite = useSelector((state) => state.app.favorite);
  if (movies.Error && !favorite) dispatch(noChangeMovie());
  const elementTitle = movies.Error ? (
    <h1>{movies.Error}</h1>
  ) : (
    <h1>{changeMovie.Title}</h1>
  );
  const elementDiscription = changeMovie
    ? Object.keys(changeMovie).map((mov) => {
        if (
          mov !== "Ratings" &&
          mov !== "Response" &&
          mov !== "Website" &&
          mov !== "Title" &&
          mov !== "Language" &&
          mov !== "Poster" &&
          mov !== "DVD" &&
          mov !== "BoxOffice" &&
          mov !== "Metascore" &&
          mov !== "Plot" &&
          mov !== "Error"
        ) {
          return (
            <p key={mov}>
              <b>{mov}: &nbsp; </b>
              {changeMovie[mov]}
            </p>
          );
        } else return null;
      })
    : null;
  return (
    <div className="container">
      <div className="description">
        {changeMovie.Year && (
          <FontAwesomeIcon
            onClick={() => dispatch(noChangeMovie())}
            className="close"
            size="2x"
            icon={close}
          />
        )}
        {elementTitle}
        {elementDiscription}
      </div>
      {changeMovie.Poster && (
        <div className="poster">
          <img
            src={changeMovie.Poster !== "N/A" ? changeMovie.Poster : noImg}
            alt=""
          ></img>
        </div>
      )}
    </div>
  );
};

export default Description;
