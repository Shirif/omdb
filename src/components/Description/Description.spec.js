import * as reactRedux from "react-redux";
import React from "react";
import Description from "./Description";

const testProps = {Year: "2000"};
const setUp = () => shallow(<Description changeMovie={testProps} />);

describe("should render Description component", () => {
  const useSelectorMock = jest.spyOn(reactRedux, "useSelector");
  const useDispatchMock = jest.spyOn(reactRedux, "useDispatch");
  let component;
  beforeEach(() => {
    useSelectorMock.mockClear();
    useDispatchMock.mockClear();
  });

  it("should contain .description-container", () => {
    const dummyDispatch = jest.fn();
    useDispatchMock.mockReturnValue(dummyDispatch);
    useSelectorMock.mockReturnValue({});
    component = setUp();
    const wrapper = component.find(".description-container");
    expect(wrapper.length).toBe(1);
  });

  it("calls click close button", () => {
    const dummyDispatch = jest.fn();
   useDispatchMock.mockReturnValue(dummyDispatch);
   expect(dummyDispatch).not.toHaveBeenCalled();
   component = setUp();
    const wrapper = component.find(".close-icon").simulate("click");
   expect(dummyDispatch).toHaveBeenCalled();

 });
 
  it("should contain Error", () => {
    const dummyDispatch = jest.fn();
   useDispatchMock.mockReturnValue(dummyDispatch);
   expect(dummyDispatch).not.toHaveBeenCalled();
   useSelectorMock.mockReturnValue({ Error: "Search movie!" });
   component = setUp();
   expect(dummyDispatch).toHaveBeenCalledWith({"type": "NO_CHANGE_MOVIE"});

 });
});
