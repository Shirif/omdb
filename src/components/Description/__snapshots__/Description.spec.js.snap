// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Description component should rander Year 1`] = `
<div
  className="container"
>
  <div
    className="description"
  >
    <FontAwesomeIcon
      border={false}
      className="close"
      fixedWidth={false}
      flip={null}
      icon={
        Object {
          "icon": Array [
            512,
            512,
            Array [],
            "f410",
            "M464 32H48C21.5 32 0 53.5 0 80v352c0 26.5 21.5 48 48 48h416c26.5 0 48-21.5 48-48V80c0-26.5-21.5-48-48-48zm0 394c0 3.3-2.7 6-6 6H54c-3.3 0-6-2.7-6-6V86c0-3.3 2.7-6 6-6h404c3.3 0 6 2.7 6 6v340zM356.5 194.6L295.1 256l61.4 61.4c4.6 4.6 4.6 12.1 0 16.8l-22.3 22.3c-4.6 4.6-12.1 4.6-16.8 0L256 295.1l-61.4 61.4c-4.6 4.6-12.1 4.6-16.8 0l-22.3-22.3c-4.6-4.6-4.6-12.1 0-16.8l61.4-61.4-61.4-61.4c-4.6-4.6-4.6-12.1 0-16.8l22.3-22.3c4.6-4.6 12.1-4.6 16.8 0l61.4 61.4 61.4-61.4c4.6-4.6 12.1-4.6 16.8 0l22.3 22.3c4.7 4.6 4.7 12.1 0 16.8z",
          ],
          "iconName": "window-close",
          "prefix": "far",
        }
      }
      inverse={false}
      listItem={false}
      mask={null}
      onClick={[Function]}
      pull={null}
      pulse={false}
      rotation={null}
      size="2x"
      spin={false}
      swapOpacity={false}
      symbol={false}
      title=""
      transform={null}
    />
    <h1 />
    <p
      key="Year"
    >
      <b>
        Year
        : Â  
      </b>
      2000
    </p>
  </div>
</div>
`;
